WcaOnRails/app/javascript/edit-schedule/EditVenue/index.jsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react'
+import { rootRender } from 'edit-schedule'
+import {
+  convertVenueActivitiesToVenueTimezone,
+  newRoomId,
+  toMicrodegrees,
+  toDegrees,
+} from '../utils'
+import { EditRoom } from './EditRoom'
+import { compose, withProps } from "recompose"
+import { withGoogleMap, GoogleMap, Marker } from "react-google-maps"
+import { Panel, Row, Col, Clearfix } from 'react-bootstrap'
+import { timezoneData } from 'wca/timezoneData.js.erb'
+
+export class EditVenue extends React.Component {
+
+  handleSinglePropertyChange = (e, propName) => {
+    this.props.venueWcif[propName] = e.target.value;
+    if (propName == "timezone") {
+      convertVenueActivitiesToVenueTimezone(this.props.venueWcif);
+    }
+    rootRender();
+  }
+
+  markerRefUpdater = (input) => this.marker = input;
+
+  handlePositionChange = () => {
+    let pos = this.marker.getPosition();
+    let newLat = toMicrodegrees(pos.lat());
+    let newLng = toMicrodegrees(pos.lng());
+    // Update parent's WCIF
+    this.props.venueWcif.latitudeMicrodegrees = newLat;
+    this.props.venueWcif.longitudeMicrodegrees = newLng;
+    rootRender();
+  }
+
+  render() {
+    let { venueWcif, index, removeVenueAction, competitionInfo } = this.props;
+    // Instead of giving *all* TZInfo, use uniq-fied rails "meaningful" subset
+    // We'll add the "country_zones" to that, because some of our competitions
+    // use TZs not included in this subset.
+    let competitionZonesKeys = Object.keys(competitionInfo.countryZones);
+    let selectKeys = _.difference(Object.keys(timezoneData), competitionZonesKeys);
+    selectKeys = _.union(competitionZonesKeys.sort(), selectKeys.sort());
+
+    let actionsHandlers = {
+      addRoom: e => {
+        e.preventDefault();
+        addRoomToVenue(venueWcif, competitionInfo);
+        rootRender();
+      },
+      removeRoom: (e, index) => {
+        e.preventDefault();
+        if (!confirm(`Are you sure you want to remove the room "${venueWcif.rooms[index].name}" and the associated schedule?`)) {
+          return;
+        }
+        venueWcif.rooms.splice(index, 1);
+        rootRender();
+      },
+    };
+
+    // Every venue col doesn't have the same height, so we need a clearfix depending on our index and viewport.
+    // In XS there is one venue per row, so no clearfix needed.
+    // In MD there are two venues per row, so if we're last, we need a clearfix
+    // In LG there are three venues per row, so if we're last, we need a clearfix
+    let clearfixes = [];
+    if (index%2 == 1) {
+      clearfixes.push(<Clearfix visibleMdBlock key={1} />);
```

Comment:
It might be possible to get rid of these Clearfix workarounds by using CSS grids. But that also might not be worth the effort =)